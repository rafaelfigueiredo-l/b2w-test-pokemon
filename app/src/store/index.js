import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
