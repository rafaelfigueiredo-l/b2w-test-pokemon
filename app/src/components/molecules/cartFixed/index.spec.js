import { Provider } from "react-redux";
import store from "../../../store";
import CartFixedContainer from "./";

describe("<CartFixedContainer />", () => {
  const component = (
    <Provider store={store}>
      <CartFixedContainer />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly CartFixedContainer component", () => {
    expect(mounted.find(".cartFixedContainer")).toHaveLength(0);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly CartFixedContainer component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
