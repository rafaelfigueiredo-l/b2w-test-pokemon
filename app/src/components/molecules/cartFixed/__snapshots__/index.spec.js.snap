// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartFixedContainer /> should mount correctly CartFixedContainer component 1`] = `""`;
